### PR TITLE
Support aiohttp 3.9.0b1

### PR DESCRIPTION
**Pull request by alexrudd2**
Thanks to @Dreamsorcerer in https://github.com/pymodbus-dev/pymodbus/issues/1883#issuecomment-1804196681

### CODE REVIEW

**Comment by janiversen** [comment]
I think we should wait !!

HA uses:
aiohttp==3.9.0b0;python_version>='3.12'
aiohttp==3.8.5;python_version<'3.12'

and that works fine, you pinned 3.9.0b0 so lets stay with that until they get a stable version.

**Comment by Dreamsorcerer** [comment]
HA is affected by a different regression (client-side), which we already have a fix for. The change in this PR is correct, for both current and previous versions of aiohttp (as shown in the docs).

File: pymodbus/server/simulator/http_server.py [hunk 1]
```diff
@@ -161,6 +161,7 @@ def __init__(
         self.serving: asyncio.Future = asyncio.Future()
         self.log_file = log_file
         self.site = None
+        self.runner: web.AppRunner
```
Comment (janiversen) [inline pymodbus/server/simulator/http_server.py:164]: This change should cause a mypy problem, because self.runner is now defined outside __init__

**Comment by Dreamsorcerer** [comment]
Even in previous versions of aiohttp, you were never running the cleanup steps for the web app.

File: pyproject.toml [hunk 2]
```diff
@@ -56,7 +56,7 @@ repl = [
     "click>=8.0.0"
 ]
 simulator = [
-    "aiohttp==3.9.0b0"
+    "aiohttp>=3.9.0b0"
```
Comment (janiversen) [inline pyproject.toml:59]: Let please not start hunting beta versions !!!!

I added the b0, but forgot to limit it to 3.12...see my preference as comment in some of the other PR´s

**Review by janiversen — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (alexrudd2) [reply pyproject.toml:59]: I'm fine with switching to 
```aiohttp==3.9.0b0;python_version>='3.12'
aiohttp==3.8.5;python_version<'3.12'
```

However, it seems orthogonal to this PR, which DreamSorcerer has stated is an improvement regardless of `aiohttp` version.
> The change in this PR is correct, for both current and previous versions of aiohttp (as shown in the docs).
> Even in previous versions of aiohttp, you were never running the cleanup steps for the web app.

File: pymodbus/server/simulator/http_server.py (hunk 1 above)
Comment (alexrudd2) [reply pymodbus/server/simulator/http_server.py:164]: `pylint` did complain... until I added this line with the type, which satisfies both `pylint` and `mypy`

File: pyproject.toml (hunk 2 above)
Comment (janiversen) [reply pyproject.toml:59]: I have no problem approving the code changes!

Comment (alexrudd2) [reply pyproject.toml]: This PR was not a version bump; in fact the opposite (expanding a pin).

Nevertheless, I have expanded the version strategy so `3.8.6` is permitted on Python <3.12.

**Review by janiversen — APPROVED** [review]
LGTM, thanks

**Pull request merged**